src/pymor/algorithms/timestepping.py

```diff
@@ -10,227 +10,619 @@
+
+    def __init__(self, num_values=None, interpolation='P1'):
+        assert not num_values or (isinstance(num_values, Number) and num_values > 1)
+        assert interpolation in self.available_interpolations
+        self.__auto_init(locals())
+
+    def solve(self, t0, t1, U0, A, F=None, M=None, mu=None, return_iter=False, return_times=False):
         """Apply time-stepper to the equation ::
 
-            M * d_t u + A(u, mu, t) = F(mu, t).
+            M * d_t u + A(u, mu, t) = F(mu, t),
+                         u(mu, t_0) = u_0(mu).
 
         Parameters
         ----------
-        initial_time
+        t0
             The time at which to begin time-stepping.
-        end_time
+        t1
             The time until which to perform time-stepping.
-        initial_data
-            The solution vector at `initial_time`.
-        operator
+        U0
+            The solution vector at `t0`.
+        A
             The |Operator| A.
-        rhs
+        F
             The right-hand side F (either |VectorArray| of length 1 or |Operator| with
             `source.dim == 1`). If `None`, zero right-hand side is assumed.
-        mass
+        M
             The |Operator| M. If `None`, the identity operator is assumed.
         mu
             |Parameter values| for which `operator` and `rhs` are evaluated. The current
             time is added to `mu` with key `t`.
-        num_values
-            The number of returned vectors of the solution trajectory. If `None`, each
-            intermediate vector that is calculated is returned.
+        return_iter
+            Determines the return data, see below.
+        return_times
+            Determines the return data, see below.
 
         Returns
         -------
-        |VectorArray| containing the solution trajectory.
+        U
+            If `return_iter == False` and `return_times == False` (the default), where `U` is a |VectorArray|
+            containing the solution trajectory.
+        (U, t)
+            If `return_iter == False` and `return_times == True`, where `t` is a list of time points corresponding to
+            `U`.
+        iterator
+            If `return_iter == True`, an iterator yielding either `U_n` (if `return_times == False`) or `(U_n, t_n)` in
+            each step.
+        """
+
+        # all checks are delegated to TimeStepperIterator
+        iterator = self.IteratorType(
+                self, t0, t1, U0, A, F=F, M=M, mu=mu, return_iter=return_iter, return_times=return_times)
+        if return_iter:
+            return iterator
+        elif return_times:
+            U = A.source.empty(reserve=self.num_values or 0)
+            t = []
+            for U_n, t_n in iterator:
+                U.append(U_n, remove_from_other=True)
+                t.append(t_n)
+            return U, t
+        else:
+            U = A.source.empty(reserve=self.num_values or 0)
+            for U_n in iterator:
+                U.append(U_n)
```

Comment:
Why is there no `remove_from_other=True` as above?

Reply:
Interestingly enough, setting `remove_from_other=True` yields an `U` of length 1 (in both places), while `False` gives the expected result. Given the code above, might this be a bug in the vectorarray?